src/qutip_cupy/__init__.py

```diff
@@ -1 +1,43 @@
+""" The qutip-cupy package provides a CuPy-based data layer for QuTiP. """
+# we need to silence this specific warning 
+# remember to remove once QuTiP moves matplotlib 
+# to an official optional dependency
+import warnings
+
+with warnings.catch_warnings():
+     warnings.filterwarnings(
+        action ='ignore',
+        category = UserWarning,
+        message = r'matplotlib not found:')
+
+     from qutip.core import data
+
 from .version import version as __version__
+
+
+
+from . import dense as cd
+CuPyDense = cd.CuPyDense
+
+data.to.add_conversions([
+     (CuPyDense, data.Dense, cd.cupydense_from_dense),
+     (data.Dense, CuPyDense, cd.dense_from_cupydense),
+ ])
+data.to.register_aliases(['cupyd'], CuPyDense)
+
+data.adjoint.add_specialisations([
+      (CuPyDense, CuPyDense, cd.cpd_adjoint),
+     ])
+data.transpose.add_specialisations([
+      (CuPyDense, CuPyDense, cd.cpd_transpose),
+     ])
+data.conj.add_specialisations([
+      (CuPyDense, CuPyDense, cd.cpd_conj),
+     ])
```

Comment:
```suggestion
data.to.add_conversions([
    (CuPyDense, data.Dense, cd.cupydense_from_dense),
    (data.Dense, CuPyDense, cd.dense_from_cupydense),
])
data.to.register_aliases(['cupyd'], CuPyDense)

data.adjoint.add_specialisations([
    (CuPyDense, CuPyDense, cd.cpd_adjoint),
])
data.transpose.add_specialisations([
    (CuPyDense, CuPyDense, cd.cpd_transpose),
])
data.conj.add_specialisations([
    (CuPyDense, CuPyDense, cd.cpd_conj),
])
```
I'm trying to tidy up the indentation a bit -- perhaps we can introduce one of the opinionated formatters like ``black`` or ``blue`` to avoid these sorts of review comments. :)

Reply:
I have pcodestyle, but till now I was just ignoring it, should have started paying attentionbefore. What do you think of adding a style check to the CI?

Reply:
I would be very much in favour of adding a style check to the CI. My vote is for pyflakes and black.